### ISSUE TITLE
Some oddities with array- and dictionary families

### ISSUE THREAD

**Issue by Wackytoaster** [Scirra/Construct-3-bugs#1604]
## Problem description

I got a bit confused when I attempted to use arrays with families (**everything also applies to dictionaries**). I think some of it might be by design, but it´s confusing regardless.

When you first create a family with a single array in it (I did this because I just wanted to get it done with the intention of adding further arrays later) and select said family in the project window, it´s properties show me the option to add an instance variable to it. Nice, just what I needed (I thought) so I added my variable and... couldn´t access it in an event (Family > compare instance variable).

Upon some investigation I saw that when I select the Family, it´s properties are NOT the familiy properties, but the array's. This means if I add an instance variable to the array, it also shows up when I select the family.

I then went ahead and duplicated the array so there are now 2 arrays inside the family. Now when I select the family it´s properties are indeed the family properties... also the option to add instance variables is now gone. I can still attempt to compare it´s instance variables in events even though it seems to not be supported for array-families to even have instance variables.

This brings me to the last point, the reason why I even need this. Because right now there is no convenient way to pick a specific array out of a family of arrays. The only picking options are the IID or the UID.

Say I have 3 arrays, A, B and C that I want to load project files into. Because I try not to hardcode too much and keep it easily expandable, I went ahead and loop through the names (stored whereever), use the name as the tag aswell as the reference to the project file (AJAX.Tag = A, file = A.json). Obviously I don´t want to now go ahead and use "on A complete, on B complete...." so instead I used "on any complete" and again loop through the names, this time comparing the AJAX.Tag.

This is where I hit the roadblock: I now know I just loaded the data for array B and I know the data loaded belongs into either A, B or C... but how do I now tell Construct to load the data into array B? I ended up using a little workaround that actually isn´t too bad, but also not as nifty as it could be with instance variables.

tl;dr

- array-families do not support instance variables, but confusingly show the option to add instance variables to it as long as there is only one array in the family
- array-families still show the option to "compare instance variables" or "set" them, despite them not supporting instance variables (or are they actually supposed to support instance variables?)
- the lack of said support for instance variables makes it a bit annoying to reference specific arrays in a way that isn´t hardcoded. Though, as mentioned, there is a workaround, I´d still prefer simply using instance variabless

## Attach a .c3p

[arrays.c3p.zip](https://github.com/Scirra/Construct-3-bugs/files/2019254/arrays.c3p.zip)
There isn´t much to see actually, it only shows the rather confusing difference in the properties of the two array-families. If you need more info on why exactly I would like instance variables on a family of arrays I can also provide a c3p for that.

## Steps to reproduce

1. Create an array
2. Add it to a new family
3. Check the families properties, you can add instance variables, yay
4. clone the array
5. Check the families properties again, no more instance variables
6. Create an event and try to "compare instance variable" of the family (it only tells you to add instance variables first :V )

## Observed result

Confusion

## Expected result

Array- and dictionary-families to clearly support instance variables (**preferred**)
or
Array- and dictionay-families to clearly **not** support instance variables (**I´d still prefer the option above**)

## Affected browsers

* Chrome: yes

## System details

*If you see a crash report dialog, please copy and paste it to where it says "PASTE HERE" below. Otherwise please go to **Menu** -> **About** -> **Platform information** and paste that information there instead.*

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 66.0.3359.181
Browser engine: Blink
Browser architecture: 64-bit
Context: webapp
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1.75
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/66.0.3359.181 Safari/537.36
C3 release: r98 (stable)
Language setting: en-US

Local storage
Storage quota (approx): 30 gb
Storage usage (approx): 414 mb (1.3%)
Persistant storage: No

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 960M Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by Wackytoaster** [comment]
Huh I think I just found out how I can do that. When I create an array there is only one in the layout, but I can also just use the "create object" action to create further arrays as needed and apply instance variables accordingly. It makes sense that it works like this. I always thought I sort of had to create an array object for each array I´m gonna use since there is no way to add several instances of them in the editor.

Regardless, I think it´s still confusing and should be clarified somehow.

**Comment by AshleyScirra** [comment]
I think this just comes down to "selecting a family of arrays in the Project Bar doesn't show the right properties". That should be fixed in the next release.
